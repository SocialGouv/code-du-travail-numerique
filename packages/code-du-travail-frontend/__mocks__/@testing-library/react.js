/* eslint-disable import/export */
import { theme } from "@socialgouv/cdtn-ui";
import { render } from "@testing-library/react";
import PropTypes from "prop-types";
import React from "react";
import { ThemeProvider } from "styled-components";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme.colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
