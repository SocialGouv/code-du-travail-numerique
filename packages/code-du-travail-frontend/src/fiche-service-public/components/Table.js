import { Table as UITable } from "@socialgouv/cdtn-ui";
import PropTypes from "prop-types";
import React from "react";

import { getText, ignoreParagraph } from "../utils.js";
import { ElementBuilder } from "./ElementBuilder.js";

const ROW_HEADER = "header";

class Table extends React.PureComponent {
  static propTypes = {
    data: PropTypes.object.isRequired,
    headingLevel: PropTypes.number.isRequired,
  };
  render() {
    const { data, headingLevel } = this.props;

    const title = data.children.find((child) => child.name === "Titre");
    const headingRows = data.children.filter(
      (child) => child.name === "Rangée" && child.attributes.type === "header"
    );
    const rows = data.children.filter(
      (child) => child.name === "Rangée" && child.attributes.type === "normal"
    );

    const columns = data.children.filter((child) => child.name === "Colonne");
    const isHeaderCell = (columnIndex) => {
      return columns[columnIndex].attributes.type === ROW_HEADER;
    };

    const handleSpan = (el) => {
      let colSpan = 1;
      let rowSpan = 1;
      if (el.attributes) {
        colSpan = el.attributes.fusionHorizontale || 1;
        rowSpan = el.attributes.fusionVerticale || 1;
      }
      return {
        colSpan,
        rowSpan,
      };
    };

    return (
      <UITable>
        {title && <caption>{getText(title)}</caption>}
        {headingRows.length > 0 && (
          <thead>
            {headingRows.map((tr, rowIndex) => (
              <tr key={rowIndex}>
                {tr.children.map((th, columnIndex) => (
                  <th key={columnIndex} {...handleSpan(th)}>
                    {th.children && (
                      <ElementBuilder
                        data={ignoreParagraph(th)}
                        headingLevel={headingLevel}
                      />
                    )}
                  </th>
                ))}
              </tr>
            ))}
          </thead>
        )}
        <tbody>
          {rows.map((tr, rowIndex) => (
            <tr key={rowIndex}>
              {tr.children.map((td, columnIndex) => {
                const Cell = isHeaderCell(columnIndex) ? "th" : "td";
                if (!td.children) {
                  return <Cell />;
                }
                return (
                  <Cell key={columnIndex} {...handleSpan(td)}>
                    <ElementBuilder
                      data={ignoreParagraph(td)}
                      headingLevel={headingLevel + 1}
                    />
                  </Cell>
                );
              })}
            </tr>
          ))}
        </tbody>
      </UITable>
    );
  }
}

export default Table;
