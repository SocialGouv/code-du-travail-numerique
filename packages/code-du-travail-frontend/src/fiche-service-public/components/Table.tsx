import React from "react";

import { getTitleInChildren, ignoreParagraph } from "../utils";
import { ElementBuilder } from "./ElementBuilder";

const ROW_HEADER = "header";

export const Table = ({
  data,
  headingLevel,
}: {
  data: any;
  headingLevel: number;
}) => {
  const title = getTitleInChildren(data);
  const headingRows = data.children.filter(
    (child) =>
      child.name === "Rangée" &&
      child.type === "element" &&
      child.attributes!.type === "header"
  );
  const rows = data.children.filter(
    (child) =>
      child.name === "Rangée" &&
      child.type === "element" &&
      child.attributes!.type === "normal"
  );

  const columns = data.children.filter((child) => child.name === "Colonne");
  const isHeaderCell = (columnIndex) => {
    return columns[columnIndex].attributes!.type === ROW_HEADER;
  };

  const handleSpan = (el) => {
    let colSpan = 1;
    let rowSpan = 1;
    if (el.attributes) {
      colSpan = el.attributes.fusionHorizontale || 1;
      rowSpan = el.attributes.fusionVerticale || 1;
    }
    return {
      colSpan,
      rowSpan,
    };
  };

  return (
    <div className="fr-table--sm fr-table fr-table">
      <div className="fr-table__wrapper">
        <div className="fr-table__container">
          <div className="fr-table__content">
            <table>
              {title && <caption>{title}</caption>}
              {headingRows.length > 0 && (
                <thead>
                  {headingRows.map((tr, rowIndex) => (
                    <tr key={rowIndex}>
                      {tr.children.map((th, columnIndex) => (
                        <th key={columnIndex} {...handleSpan(th)}>
                          {th.children && (
                            <ElementBuilder
                              data={ignoreParagraph(th)}
                              headingLevel={headingLevel}
                            />
                          )}
                        </th>
                      ))}
                    </tr>
                  ))}
                </thead>
              )}
              <tbody>
                {rows.map((tr, rowIndex) => (
                  <tr key={rowIndex}>
                    {tr.children.map((td, columnIndex) => {
                      const Cell = isHeaderCell(columnIndex) ? "th" : "td";
                      if (!td.children) {
                        return <Cell key={columnIndex} />;
                      }
                      return (
                        <Cell key={columnIndex} {...handleSpan(td)}>
                          <ElementBuilder
                            data={ignoreParagraph(td)}
                            headingLevel={headingLevel + 1}
                          />
                        </Cell>
                      );
                    })}
                  </tr>
                ))}
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Table;
