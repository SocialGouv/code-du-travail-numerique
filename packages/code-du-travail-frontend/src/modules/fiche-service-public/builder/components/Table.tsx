import React from "react";

import { getInChildrenByName, getText, ignoreParagraph } from "../utils";
import { ElementBuilder } from "./ElementBuilder";
import { FicheSPDataTableau } from "../type";

const ROW_HEADER = "header";

export const Table = ({
  data,
  headingLevel,
}: {
  data: FicheSPDataTableau;
  headingLevel: number;
}) => {
  const title = getInChildrenByName(data, "Titre");
  const headingRows = data.children.filter(
    (child) => child.name === "Rangée" && child.attributes.type === "header"
  );
  const rows = data.children.filter(
    (child) => child.name === "Rangée" && child.attributes.type === "normal"
  );

  const columns = data.children.filter((child) => child.name === "Colonne");
  const isHeaderCell = (columnIndex) => {
    return columns[columnIndex].attributes.type === ROW_HEADER;
  };

  const handleSpan = (el) => {
    let colSpan = 1;
    let rowSpan = 1;
    if (el.attributes) {
      colSpan = el.attributes.fusionHorizontale || 1;
      rowSpan = el.attributes.fusionVerticale || 1;
    }
    return {
      colSpan,
      rowSpan,
    };
  };

  return (
    <div className="fr-table--sm fr-table fr-table">
      <div className="fr-table__wrapper">
        <div className="fr-table__container">
          <div className="fr-table__content">
            <table>
              {title && <caption>{getText(title)}</caption>}
              {headingRows.length > 0 && (
                <thead>
                  {headingRows.map((tr, rowIndex) => (
                    <tr key={rowIndex}>
                      {tr.children.map((th, columnIndex) => (
                        <th key={columnIndex} {...handleSpan(th)}>
                          {th.children && (
                            <ElementBuilder
                              data={ignoreParagraph(th)}
                              headingLevel={headingLevel}
                            />
                          )}
                        </th>
                      ))}
                    </tr>
                  ))}
                </thead>
              )}
              <tbody>
                {rows.map((tr, rowIndex) => (
                  <tr key={rowIndex}>
                    {tr.children.map((td, columnIndex) => {
                      const Cell = isHeaderCell(columnIndex) ? "th" : "td";
                      if (!td.children) {
                        return <Cell key={columnIndex} />;
                      }
                      return (
                        <Cell key={columnIndex} {...handleSpan(td)}>
                          <ElementBuilder
                            data={ignoreParagraph(td)}
                            headingLevel={headingLevel + 1}
                          />
                        </Cell>
                      );
                    })}
                  </tr>
                ))}
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Table;
