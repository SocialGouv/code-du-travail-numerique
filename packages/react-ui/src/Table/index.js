import PropTypes from "prop-types";
import React from "react";
import styled, { css } from "styled-components";

import { OverflowWrapper } from "../OverflowWrapper/index.js";
import { box, spacings } from "../theme";

export const Table = ({ children, ...props }) => (
  <OverflowWrapper>
    <StyledTable {...props}>{children}</StyledTable>
  </OverflowWrapper>
);

Table.propTypes = {
  children: PropTypes.node.isRequired,
};

const StyledTable = styled.table`
  text-align: left;
  empty-cells: show;
  background-color: ${({ theme }) => theme.white};
  border: ${({ theme }) => box.border(theme.border)};
  border-collapse: collapse;
  border-spacing: 0;

  caption {
    padding: ${spacings.small} 0;
    font-style: italic;
    text-align: center;
  }

  ${({ theme, stripes }) => {
    if (stripes) {
      return css`
        tr:nth-child(even) {
          background-color: ${theme.bgSecondary};
        }
      `;
    }
  }}

  td,
  th {
    padding: ${spacings.small} ${spacings.base};
    border: ${({ theme }) => box.border(theme.border)};
  }

  th {
    color: ${({ theme }) => theme.title};
    font-weight: 600;
    background: ${({ theme }) => theme.bgTertiary};
  }

  thead,
  tfoot {
    padding: ${spacings.small} ${spacings.base};
    vertical-align: bottom;
    background: ${({ theme }) => theme.bgSecondary};
  }

  tfoot {
    vertical-align: top;
  }
`;
