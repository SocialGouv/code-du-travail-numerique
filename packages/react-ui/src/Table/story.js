import React from "react";

import { Section } from "../layout/Section/index";
import { Table } from "./index.js";

export default {
  component: Table,
  title: "Components/Table",
};

export const base = () => (
  <>
    <Section>
      <Table>
        <caption>Some caption</caption>
        <thead>
          <tr>
            <th>Table header 1</th>
            <th>Table header 2</th>
            <th>Table header 3</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <th>Line 1</th>
            <td>some data</td>
            <td>some data</td>
          </tr>
          <tr>
            <th>Line 2</th>
            <td>some data</td>
            <td>some data</td>
          </tr>
        </tbody>
      </Table>
    </Section>
    <Section>
      <Table stripes>
        <caption>Table with stripes</caption>
        <thead>
          <tr>
            <th>Table header 1</th>
            <th>Table header 2</th>
            <th>Table header 3</th>
            <th>Table header 4</th>
            <th>Table header 5</th>
            <th>Table header 6</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <th>Line 1</th>
            <td>some data</td>
            <td>some data</td>
            <td>some data</td>
            <td>some data</td>
            <td>some data</td>
          </tr>
          <tr>
            <th>Line 2</th>
            <td>some data</td>
            <td>some data</td>
            <td>some data</td>
            <td>some data</td>
            <td>some data</td>
          </tr>
          <tr>
            <th>Line 3</th>
            <td>some data</td>
            <td>some data</td>
            <td>some data</td>
            <td>some data</td>
            <td>some data</td>
          </tr>
          <tr>
            <th>Line 4</th>
            <td>some data</td>
            <td>some data</td>
            <td>some data</td>
            <td>some data</td>
            <td>some data</td>
          </tr>
          <tr>
            <th>Line 2</th>
            <td>some data</td>
            <td>some data</td>
            <td>some data</td>
            <td>some data</td>
            <td>some data</td>
          </tr>
        </tbody>
      </Table>
    </Section>
  </>
);
