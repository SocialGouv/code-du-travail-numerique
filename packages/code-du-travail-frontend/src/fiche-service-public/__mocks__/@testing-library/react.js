import React from "react";
import PropTypes from "prop-types";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "@socialgouv/cdtn-ui";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme.colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
