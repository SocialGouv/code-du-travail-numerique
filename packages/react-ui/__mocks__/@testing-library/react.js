import { render } from "@testing-library/react";
import PropTypes from "prop-types";
import React from "react";
import { ThemeProvider } from "styled-components";

import { colors } from "../../src/theme";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
