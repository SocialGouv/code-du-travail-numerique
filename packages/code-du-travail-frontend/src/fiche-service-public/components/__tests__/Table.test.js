import React from "react";
import { render } from "@testing-library/react";
import Table from "../Table";
import tableDataMock from "./mocks/tableData.json";

describe("<Table />", () => {
  it("should render with th, td, colspan etc.", () => {
    const { container } = render(
      <Table data={tableDataMock} headingLevel={2} />,
    );
    expect(container).toMatchSnapshot();
  });
});
