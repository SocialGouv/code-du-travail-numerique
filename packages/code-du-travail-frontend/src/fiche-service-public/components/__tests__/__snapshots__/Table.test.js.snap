// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> should render with th, td, colspan etc. 1`] = `
.c0 {
  position: relative;
  overflow: hidden;
}

.c0:before,
.c0:after {
  position: absolute;
  top: 0;
  bottom: 0;
  z-index: 1;
  display: block;
  width: 4rem;
  opacity: 0;
  -webkit-transition: opacity 0.3s linear;
  transition: opacity 0.3s linear;
  content: "";
  pointer-events: none;
  background: radial-gradient( ellipse at center, #fff 15%, rgba(255,255,255,0) 80% );
}

.c0:before {
  left: -2rem;
}

.c0:after {
  right: -2rem;
}

.c1 {
  overflow-x: auto;
}

.c2 {
  text-align: left;
  empty-cells: show;
  background-color: #fff;
  border: 1px solid #bbcadf;
  border-collapse: collapse;
  border-spacing: 0;
}

.c2 caption {
  padding: 1rem 0;
  font-style: italic;
  text-align: center;
}

.c2 td,
.c2 th {
  padding: 1rem 1.6rem;
  border: 1px solid #bbcadf;
}

.c2 th {
  color: #2f3b6c;
  font-weight: 600;
  background: #e4e8ef;
}

.c2 thead,
.c2 tfoot {
  padding: 1rem 1.6rem;
  vertical-align: bottom;
  background: #f2f5fa;
}

.c2 tfoot {
  vertical-align: top;
}

@media (max-width:600px) {
  .c0 {
    overflow-x: hidden;
  }
}

@media (max-width:600px) {
  .c0:before,
  .c0:after {
    display: block;
  }
}

@media (max-width:600px) {
  .c1 {
    overflow-x: auto;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <table
        class="c2"
      >
        <caption>
          Conditions ouvrant droit à la retraite anticipée pour carrière longue
        </caption>
        <thead>
          <tr>
            <th
              colspan="1"
              rowspan="2"
            >
              Année de naissance
            </th>
            <th
              colspan="1"
              rowspan="2"
            >
              Âge de départ à la retraite envisagé (à partir de)
            </th>
            <th
              colspan="1"
              rowspan="2"
            >
              Durée d'assurance minimale cotisée (en trimestres)
            </th>
            <th
              colspan="2"
              rowspan="1"
            >
              Durée d'assurance minimale en début de carrière
            </th>
          </tr>
          <tr>
            <th
              colspan="1"
              rowspan="1"
            >
              Si vous êtes né entre janvier et septembre
            </th>
            <th
              colspan="1"
              rowspan="1"
            >
              Si vous êtes né entre octobre et décembre
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <th
              colspan="1"
              rowspan="2"
            >
              1959
            </th>
            <td
              colspan="1"
              rowspan="1"
            >
              57 ans et 8 mois
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              175
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              <strong>
                5
              </strong>
               trimestres à la fin de l'année des 
              <strong>
                16 ans
              </strong>
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              <strong>
                4
              </strong>
               trimestres à la fin de l'année des 
              <strong>
                16 ans
              </strong>
            </td>
          </tr>
          <tr>
            <th
              colspan="1"
              rowspan="1"
            >
              60 ans
            </th>
            <td
              colspan="1"
              rowspan="1"
            >
              167
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              <strong>
                5
              </strong>
               trimestres à la fin de l'année des 
              <strong>
                20 ans
              </strong>
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              <strong>
                4
              </strong>
               trimestres à la fin de l'année des 
              <strong>
                20 ans
              </strong>
            </td>
          </tr>
          <tr>
            <th
              colspan="1"
              rowspan="2"
            >
              1960
            </th>
            <td
              colspan="1"
              rowspan="1"
            >
              58 ans
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              175
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              <strong>
                5
              </strong>
               trimestres à la fin de l'année des 
              <strong>
                16 ans
              </strong>
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              <strong>
                4
              </strong>
               trimestres à la fin de l'année des 
              <strong>
                16 ans
              </strong>
            </td>
          </tr>
          <tr>
            <th
              colspan="1"
              rowspan="1"
            >
              60 ans
            </th>
            <td
              colspan="1"
              rowspan="1"
            >
              167
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              <strong>
                5
              </strong>
               trimestres à la fin de l'année des 
              <strong>
                20 ans
              </strong>
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              <strong>
                4
              </strong>
               trimestres à la fin de l'année des 
              <strong>
                20 ans
              </strong>
            </td>
          </tr>
          <tr>
            <th
              colspan="1"
              rowspan="2"
            >
              1961, 1962 ou 1963
            </th>
            <td
              colspan="1"
              rowspan="1"
            >
              58 ans
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              176
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              <strong>
                5
              </strong>
               trimestres à la fin de l'année des 
              <strong>
                16 ans
              </strong>
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              <strong>
                4
              </strong>
               trimestres à la fin de l'année des 
              <strong>
                16 ans
              </strong>
            </td>
          </tr>
          <tr>
            <th
              colspan="1"
              rowspan="1"
            >
              60 ans
            </th>
            <td
              colspan="1"
              rowspan="1"
            >
              168
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              <strong>
                5
              </strong>
               trimestres à la fin de l'année des 
              <strong>
                20 ans
              </strong>
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              <strong>
                4
              </strong>
               trimestres à la fin de l'année des 
              <strong>
                20 ans
              </strong>
            </td>
          </tr>
          <tr>
            <th
              colspan="1"
              rowspan="2"
            >
              1964, 1965 ou 1966
            </th>
            <td
              colspan="1"
              rowspan="1"
            >
              58 ans
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              177
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              <strong>
                5
              </strong>
               trimestres à la fin de l'année des 
              <strong>
                16 ans
              </strong>
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              <strong>
                4
              </strong>
               trimestres à la fin de l'année des 
              <strong>
                16 ans
              </strong>
            </td>
          </tr>
          <tr>
            <th
              colspan="1"
              rowspan="1"
            >
              60 ans
            </th>
            <td
              colspan="1"
              rowspan="1"
            >
              169
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              <strong>
                5
              </strong>
               trimestres à la fin de l'année des 
              <strong>
                20 ans
              </strong>
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              <strong>
                4
              </strong>
               trimestres à la fin de l'année des 
              <strong>
                20 ans
              </strong>
            </td>
          </tr>
          <tr>
            <th
              colspan="1"
              rowspan="2"
            >
              1967, 1968 ou 1969
            </th>
            <td
              colspan="1"
              rowspan="1"
            >
              58 ans
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              178
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              <strong>
                5
              </strong>
               trimestres à la fin de l'année des 
              <strong>
                16 ans
              </strong>
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              <strong>
                4
              </strong>
               trimestres à la fin de l'année des 
              <strong>
                16 ans
              </strong>
            </td>
          </tr>
          <tr>
            <th
              colspan="1"
              rowspan="1"
            >
              60 ans
            </th>
            <td
              colspan="1"
              rowspan="1"
            >
              170
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              <strong>
                5
              </strong>
               trimestres à la fin de l'année des 
              <strong>
                20 ans
              </strong>
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              <strong>
                4
              </strong>
               trimestres à la fin de l'année des 
              <strong>
                20 ans
              </strong>
            </td>
          </tr>
          <tr>
            <th
              colspan="1"
              rowspan="2"
            >
              1970, 1971 ou 1972
            </th>
            <td
              colspan="1"
              rowspan="1"
            >
              58 ans
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              179
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              <strong>
                5
              </strong>
               trimestres à la fin de l'année des 
              <strong>
                16 ans
              </strong>
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              <strong>
                4
              </strong>
               trimestres à la fin de l'année des 
              <strong>
                16 ans
              </strong>
            </td>
          </tr>
          <tr>
            <th
              colspan="1"
              rowspan="1"
            >
              60 ans
            </th>
            <td
              colspan="1"
              rowspan="1"
            >
              171
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              <strong>
                5
              </strong>
               trimestres à la fin de l'année des 
              <strong>
                20 ans
              </strong>
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              <strong>
                4
              </strong>
               trimestres à la fin de l'année des 
              <strong>
                20 ans
              </strong>
            </td>
          </tr>
          <tr>
            <th
              colspan="1"
              rowspan="2"
            >
              À partir de 1973
            </th>
            <td
              colspan="1"
              rowspan="1"
            >
              58 ans
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              180
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              <strong>
                5
              </strong>
               trimestres à la fin de l'année des 
              <strong>
                16 ans
              </strong>
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              <strong>
                4
              </strong>
               trimestres à la fin de l'année des 
              <strong>
                16 ans
              </strong>
            </td>
          </tr>
          <tr>
            <th
              colspan="1"
              rowspan="1"
            >
              60 ans
            </th>
            <td
              colspan="1"
              rowspan="1"
            >
              172
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              <strong>
                5
              </strong>
               trimestres à la fin de l'année des 
              <strong>
                20 ans
              </strong>
            </td>
            <td
              colspan="1"
              rowspan="1"
            >
              <strong>
                4
              </strong>
               trimestres à la fin de l'année des 
              <strong>
                20 ans
              </strong>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
